File: 01731.cpp
#include <iostream>
using namespace std;

int main() {
	int N; cin >> N;
	
	int a, P, Qp, Qm; cin >> P >> a;
	Qp = a - P; Qm = a / P;
	P = a;

	bool plus = true;
	for (int i = 0; i < N-2; i++) {
		cin >> a;

		if (a - P != Qp) plus = false;
		P = a;
	}

	if (plus) cout << P + Qp << endl;
	else cout << P * Qm << endl;
}

File: 01731.py
N = int(input())

P = int(input())
a = int(input())
Qp = a - P; Qm = a // P
P = a

plus = True
for i in range(N - 2):
    a = int(input())

    if a - P != Qp:
        plus = False

    P = a

if plus: print(P + Qp)
else: print(P * Qm)
